webpack/scenes/ContentViews/Details/ComponentContentViews/ComponentContentViewAddModal.js

```diff
@@ -0,0 +1,153 @@
+import PropTypes from 'prop-types';
+import { Modal, ModalVariant, FormSelect,
+  FormSelectOption, Checkbox, Form, FormGroup,
+  ActionGroup, Button } from '@patternfly/react-core';
+import { useDispatch, useSelector, shallowEqual } from 'react-redux';
+import { translate as __ } from 'foremanReact/common/I18n';
+import { STATUS } from 'foremanReact/constants';
+import { selectCVDetails,
+  selectCVDetailStatus,
+  selectCVDetailError } from '../../Details/ContentViewDetailSelectors';
+import { addComponent } from '../ContentViewDetailActions';
+
+const ComponentContentViewAddModal = ({
+  cvId, componentCvId, componentId, latest, show, setIsOpen,
+}) => {
+  const dispatch = useDispatch();
+  const componentDetails = useSelector(
+    state => selectCVDetails(state, componentCvId),
+    shallowEqual,
+  );
+  const componentStatus = useSelector(
+    state => selectCVDetailStatus(state, componentCvId),
+    shallowEqual,
+  );
+  const componentError = useSelector(
+    state => selectCVDetailError(state, componentCvId),
+    shallowEqual,
+  );
+  const [cvName, setCvName] = useState('');
+  const [options, setOptions] = useState([]);
+  const [formLatest, setFormLatest] = useState(componentId ? latest : false);
+  const [selected, setSelected] = useState(null);
+  const versionsLoading = componentStatus === STATUS.PENDING;
+
+  useEffect(() => {
+    if (!versionsLoading && componentDetails) {
+      const { name, versions } = componentDetails;
+      const versionMutable = versions;
+      setCvName(name);
+      const opt = versionMutable.map(item => ({ value: item.id, label: __(`Version ${item.version}`) }));
+      setOptions([...opt].reverse());
+      setSelected(opt.slice(-1)[0].value);
+    }
+  }, [componentDetails, componentStatus, componentError, versionsLoading]);
+
+  const getAddParams = () => {
+    if (formLatest) {
+      return [{ latest: true, content_view_id: componentCvId }];
+    }
+    return [{ content_view_version_id: selected }];
+  };
+
+  const getUpdateParams = () => {
+    if (formLatest) {
+      return { id: componentId, latest: formLatest, compositeContentViewId: cvId };
+    }
+    return { id: componentId, compositeContentViewId: cvId, content_view_version_id: selected };
+  };
+
+  const updateLatest = (checked) => {
+    setFormLatest(checked);
+    if (checked) setSelected(options[0]);
+  };
+
+  const onSubmit = () => {
+    if (componentId) {
+      dispatch(addComponent({
+        compositeContentViewId: cvId,
+        components: getUpdateParams(),
+      }));
+    } else {
+      dispatch(addComponent({
+        compositeContentViewId: cvId,
+        components: getAddParams(),
+      }));
+    }
+    setIsOpen(false);
+  };
+
+  if (show && versionsLoading) return null;
+
+  return (
+    <Modal
+      title={componentId ? __('Update version') : __('Add component')}
+      variant={ModalVariant.small}
+      isOpen={show}
+      description={__(`Select available version of ${cvName} to add`)}
```

Comment:
Changing the wording here will avoid having to change this text based on if we are adding or updating the version:
```suggestion
      description={__(`Select available version of ${cvName} to use`)}
```

Reply:
Updated! 👍🏽